modules/core/src/copy.cpp

```diff
@@ -564,24 +564,292 @@ Mat& Mat::setTo(InputArray _value, InputArray _mask)
+{
+#if CV_SSSE3
+    static const __m128i perm = _mm_setr_epi8(14, 15, 12, 13, 10, 11, 8, 9, 6, 7, 4, 5, 2, 3, 0, 1);
+    return v_uint8x16(_mm_shuffle_epi8(_mm_loadu_si128((__m128i*)(ptr)), perm));
+#elif CV_VSX
+    static const vec_uchar16 perm = {14, 15, 12, 13, 10, 11, 8, 9, 6, 7, 4, 5, 2, 3, 0, 1};
+    vec_uchar16 vec = vsx_ld(0, ptr);
+    return v_uint8x16(vec_perm(vec, vec, perm));
+#elif CV_NEON
+    uint16x8_t vec = vrev64q_u16(vld1q_u16((ushort*)ptr));
+    return v_reinterpret_as_u8(v_uint16x8(vextq_u16(vec, vec, 4)));
+#else
+    return v_uint8x16(*(ptr + v_uint8x16::nlanes - 2),  *(ptr + v_uint8x16::nlanes - 1),
+                      *(ptr + v_uint8x16::nlanes - 4),  *(ptr + v_uint8x16::nlanes - 3),
+                      *(ptr + v_uint8x16::nlanes - 6),  *(ptr + v_uint8x16::nlanes - 5),
+                      *(ptr + v_uint8x16::nlanes - 8),  *(ptr + v_uint8x16::nlanes - 7),
+                      *(ptr + v_uint8x16::nlanes - 10), *(ptr + v_uint8x16::nlanes - 9),
+                      *(ptr + v_uint8x16::nlanes - 12), *(ptr + v_uint8x16::nlanes - 11),
+                      *(ptr + v_uint8x16::nlanes - 14), *(ptr + v_uint8x16::nlanes - 13),
+                      *(ptr + v_uint8x16::nlanes - 16), *(ptr + v_uint8x16::nlanes - 15));
+#endif
+}
+
+CV_ALWAYS_INLINE v_uint8x16 v_load_mirror_4( const uchar* ptr )
+{
+#if CV_SSE2
+    return v_uint8x16(_mm_shuffle_epi32(_mm_loadu_si128((__m128i*)(ptr)), (3 << 0) | (2 << 2) | (1 << 4) | (0 << 6)));
+#elif CV_VSX
+    static const vec_uchar16 perm = {12, 13, 14, 15, 8, 9, 10, 11, 4, 5, 6, 7, 0, 1, 2, 3};
+    vec_uchar16 vec = vsx_ld(0, ptr);
+    return v_uint8x16(vec_perm(vec, vec, perm));
+#elif CV_NEON
+    uint32x4_t vec = vrev64q_u32(vld1q_u32((uint*)ptr));
+    return v_reinterpret_as_u8(v_uint32x4(vextq_u32(vec, vec, 2)));
+#else
+    return v_uint8x16(*(ptr + v_uint8x16::nlanes - 4),  *(ptr + v_uint8x16::nlanes - 3),
+                      *(ptr + v_uint8x16::nlanes - 2),  *(ptr + v_uint8x16::nlanes - 1),
+                      *(ptr + v_uint8x16::nlanes - 8),  *(ptr + v_uint8x16::nlanes - 7),
+                      *(ptr + v_uint8x16::nlanes - 6),  *(ptr + v_uint8x16::nlanes - 5),
+                      *(ptr + v_uint8x16::nlanes - 12), *(ptr + v_uint8x16::nlanes - 11),
+                      *(ptr + v_uint8x16::nlanes - 10), *(ptr + v_uint8x16::nlanes - 9),
+                      *(ptr + v_uint8x16::nlanes - 16), *(ptr + v_uint8x16::nlanes - 15),
+                      *(ptr + v_uint8x16::nlanes - 14), *(ptr + v_uint8x16::nlanes - 13));
+#endif
+}
+
+CV_ALWAYS_INLINE v_uint8x16 v_load_mirror_8( const uchar* ptr )
+{
+#if CV_SSE2
+    return v_uint8x16(_mm_shuffle_epi32(_mm_loadu_si128((__m128i*)(ptr)), (2 << 0) | (3 << 2) | (0 << 4) | (1 << 6)));
+#elif CV_VSX
+    static const vec_uchar16 perm = {8, 9, 10, 11, 12, 13, 14, 15, 0, 1, 2, 3, 4, 5, 6, 7};
+    vec_uchar16 vec = vsx_ld(0, ptr);
+    return v_uint8x16(vec_perm(vec, vec, perm));
```

Comment:
Is this the same as xxswapd (e.g vec_permxxdi(vec,vec,2))?

Reply:
Outdated code.